src/nvim/api/vim.c

```diff
@@ -55,6 +56,42 @@ void nvim_command(String command, Error *err)
   try_end(err);
 }
 
+/// Retrieves highlight description from its name
+///
+/// @param name Highlight group name
+/// @return a highlight description e.g. {'bold': true, 'bg': 123, 'fg': 42}
+/// @see nvim_get_hl_by_id
+Dictionary nvim_get_hl_by_name(String name, Error *err)
+  FUNC_API_SINCE(3)
+{
+  Dictionary result = ARRAY_DICT_INIT;
+  int id = syn_name2id((const char_u *)name.data);
+
+  if (id == 0) {
+    api_set_error(err, kErrorTypeException, "Invalid highlight name %s",
+                  name.data);
+    return result;
+  }
+  result = nvim_get_hl_by_id(id, err);
+  return result;
+}
+
+/// Retrieves highlight description from its id
+///
+/// @param hl_id highlight id as returned by hlID()
```

Comment:
putting bars around tags (`|hlID()|`) will link the tag when the API docs are autogenerated.